jetstream/publish.go

```diff
@@ -415,6 +453,10 @@ func (js *jetStream) handleAsyncReply(m *nats.Msg) {
 		}
 	}
 
+	if paf.timeout != nil {
+		paf.timeout.Stop()
```

Comment:
from Go v1.23 onwards the timers are GC without explicit stop but do we need to add something extra to cleanup publishers to remove these too when called? https://github.com/nats-io/nats.go/blob/main/js.go#L766

Reply:
I think what we have to do is make sure we don't try to resolve timeout after acquiring the lock if it's already been resolved elsewhere. I added that to `time.AfterFunc` now:
```go
paf.timeout = time.AfterFunc(js.publisher.ackTimeout, func() {
	js.publisher.Lock()
	defer js.publisher.Unlock()

	if _, ok := js.publisher.acks[id]; !ok {
		// paf has already been resolved
		// while waiting for the lock
		return
	}
```